src/DotLiquid/StandardFilters.cs

```diff
@@ -706,49 +706,73 @@ public static object Minus(Context context, object input, object operand)
         /// <param name="places">Number of decimal places for rounding</param>
-        /// <returns>The rounded value; null if an exception have occurred</returns>
-        public static object Round(object input, object places = null)
+        /// <returns>The rounded value; zero if input is invalid, or rounded to 0 decimals if places is invalid</returns>
+        /// <remarks>Behaviour differs from Ruby implementation for negative places values.
+        /// This will treat it as any other invalid places value, and round to closest integer.</remarks>
+        [LiquidFilter(MinVersion = SyntaxCompatibility.DotLiquid24)]
+        public static object Round(Context context, object input, object places = null)
         {
-            try
-            {
-                var p = places == null ? 0 : Convert.ToInt32(places);
-                var i = Convert.ToDecimal(input);
-                return Math.Round(i, p);
-            }
-            catch (Exception)
+            int decimals = 0;
+            if (places != null)
             {
-                return null;
+                decimal placesValue = places.CoerceToDecimal(context.FormatProvider, 0m);
+                const decimal MinDecimalPlaces = 0m;
+                const decimal MaxDecimalPlaces = 28m;
+                placesValue = Math.Max(MinDecimalPlaces, Math.Min(MaxDecimalPlaces, placesValue));
+                decimals = (int)Math.Floor(placesValue);
             }
+
+            decimal inputValue = input.CoerceToDecimal(context.FormatProvider, 0m);
+            return Math.Round(inputValue, decimals);
         }
 
         /// <summary>
         /// Rounds a decimal value up to the next integer, unless already the integer value, removing all decimal places 
         /// </summary>
         /// <param name="context">The DotLiquid context</param>
         /// <param name="input">Input to be transformed by this filter</param>
-        /// <returns>The rounded value; null if an exception have occurred</returns>
+        /// <returns>The rounded value; zero if an exception has occurred</returns>
+        [LiquidFilter(MinVersion = SyntaxCompatibility.DotLiquid24)]
         public static object Ceil(Context context, object input)
         {
-            if (decimal.TryParse(input.ToString(), NumberStyles.Any, context.CurrentCulture, out decimal d))
-                return Math.Ceiling(d);
-            else
-                return null;
+            if (input == null) return 0;
+
+            if (input is string inputString)
+            {
+                input = inputString.CoerceToNumericType(context.FormatProvider, 0);
+            }
+
+            if (input is decimal inputDecimal) { return Math.Ceiling(inputDecimal); }
+            else if (input is double inputDouble) { return Math.Ceiling(inputDouble); }
+            else if (input is int inputInt32) { return inputInt32; }
+            else if (input is long inputInt64) { return inputInt64; }
+            else return 0;

```

Comment:
what happens to uint or the other types defined as IsNumeric?

Reply:
Good point. I am only handling the types that can be returned from CoerceToNumeric, but obviously they could pass in something else.